src/xapi_schema/spec.cljc

```diff
@@ -239,20 +233,39 @@
       (sgen/uuid))))
 
 
+;; Currently ignores leap seconds
+(defn- valid-timestamp?
+  [timestamp]
+  (let [[ts year month day _hour _min _sec _sec-frac _offset]
+        (re-matches TimestampRegEx timestamp)]
+    (cond
+      (nil? ts) ; fails regex
+      false
+      (= "02" month)
+      (let [year-int #?(:clj (Integer/parseInt year) :cljs (js/parseInt year))]
+        (if (or (and (= 0 (mod year-int 4)) (not= 0 (mod year-int 100)))
+                (= 0 (mod year-int 400)))
+          (not (#{"30" "31"} day)) ; leap year
+          (not (#{"29" "30" "31"} day))))
```

Comment:
might be better to do a numeric comparison opposed to set matching.
`[day-int #?(:clj (Integer/parseInt day) :cljs (js/parseInt day))]`
....
`(not (>= 30 day-int)`
`(not (>= 29 day-int)`

^^^ method also applies to month = 4 / 6 / 9 / 11